distribution/push_v2.go

```diff
@@ -337,6 +337,10 @@ func (pd *v2PushDescriptor) Upload(ctx context.Context, progressOutput progress.
 		// unable to mount layer from this repository, so this source mapping is no longer valid
 		logrus.Debugf("unassociating layer %s (%s) with %s", diffID, mountFrom.Digest, mountFrom.SourceRepository)
 		pd.v2MetadataService.Remove(mountFrom)
+		if err != nil {
+			// return the raw error so we can retry uploading this layer without a mount
+			return err
+		}
```

Comment:
Updated to retry multiple blob mounts, falling back to a standard blob upload in a loop